### PR TITLE
Move custom ui-menu to core api and update user menu

### LINKED ISSUE #1206: User menu cut on wider resolutions
## Summary

User menu gets cut off on some wider resolutions.

## System information

 - Version: 0.4.0alpha1.dev352
 - Operating system: Linux
 - Browser: Chrome & FF

## Screenshots

kolibri - mozilla firefox_149

kolibri - google chrome_145

manage users - kolibri - google chrome_144

manage users - kolibri - google chrome_143

OK here:

kolibri - google chrome_146

### PR DESCRIPTION
**Pull request by christianmemije**
# Details

### Summary

* Adds the custom ui-menu we were using to the core api for DRYness
* Replaces the dropdown menu with a our own simple implementation 
* ~TODO: Need to style the `custom-ui-menu`, stay tuned~ Done

localhost_8000_learn_

### Reviewer guidance

* Test on different viewports
* Test keyboard support.

### References

* FIxes #1206 
* https://github.com/fle-internal/kolibri-instant-schools-plugin/pull/138 depends on this 

# Contributor Checklist

- [x] PR has the correct target milestone
- [x] PR has the appropriate labels
- [x] Changes in the PR do not introduce accessibility regressions ([confimed by testing with one of the recommended tools](http://kolibri.readthedocs.io/en/develop/dev/manual_testing.html#accessibility-a11y-testing))  
- [x] If PR is ready for review, it has been assigned or requests review from someone
- [x] ~Documentation is updated as necessary~ N/A
- [x] ~External dependency files are updated (`yarn` and `pip`)~ N/A
- [x] ~If internal dependency is updated, link to diff is included~ N/A
- [ ] Screenshots of any front-end changes are in the PR description
- [x] ~CHANGELOG.rst is updated for high-level changes~ N/A
- [x] ~You've added yourself to AUTHORS.rst if you're not there~ N/A

# Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] PR has been fully tested manually

### CODE REVIEW

**Comment by indirectlylit** [comment]
@christianmemije I haven't tested this yet, but one thing to look out for is RTL compatibility

**Comment by christianmemije** [comment]
@indirectlylit Looks good to me? 
rtl

**Comment by radinamatic** [comment]
@christianmemije How do I switch to RTL to test this on pex? There doesn't seem to be available a RTL language code for the `language setdefault` command:
```
osboxes@osboxes:~/Desktop$ ./kolibri-0.6.1.dev020171117014028-git.pex language setdefault ar-ar
WARNING:root:Invalid language code ar-ar. Must be one of: [u'en', u'sw-tz', u'es-es', u'es-mx', u'fr-fr', u'pt-pt', u'hi-in']
osboxes@osboxes:~/Desktop$ 
```

**Comment by christianmemije** [comment]
@radinamatic No need to test RTL, let's just focus on different devices.

**Comment by christianmemije** [comment]
Plus no fancy css, all there is to it is:
```css
  .user-menu-dropdown
     position: fixed
     right: 0
```

File: kolibri/core/assets/src/views/custom-ui-menu/menu-option.vue [hunk 1]
```diff
@@ -2,7 +2,6 @@
 
   <li
     class="ui-menu-option"
-    ref="menuOption"
     role="menu-item"
```
Comment (radinamatic) [inline kolibri/core/assets/src/views/custom-ui-menu/menu-option.vue]: Role should be `menuitem`, not `menu-item`.

ubuntu3 start running - oracle vm virtualbox_706

**Comment by radinamatic** [comment]
Position of the menu seems to be correct in both FF & Chrome on Linux, on various screen widths:

### Firefox

ubuntu3 start running - oracle vm virtualbox_701

### Chrome

ubuntu3 start running - oracle vm virtualbox_703

ubuntu3 start running - oracle vm virtualbox_704

As far as keyboard navigation goes, the menu does open with both `Space` and `Enter` as expected, but according to [WAI-ARIA practices](https://www.w3.org/TR/wai-aria-practices/#menu), the navigation between `menuitems` should be possible with arrow keys as well as with `Tab/Shift+Tab` keys.

**Comment by christianmemije** [comment]
@radinamatic That is an issue of the `ui-menu`. Could you open a separate issue for that? :pray:

**Comment by radinamatic** [comment]
@christianmemije  Done! #2634

**Comment by indirectlylit** [comment]
@radinamatic I pushed a commit re the 'role'. Mind checking and approving the changes?

**Pull request merged**